machine/machine_core/ssh_connection.py

```diff
@@ -249,7 +249,7 @@ def _ensure_ssh_master(self):
             self._start_ssh_master()
 
     def __ssh_direct_opt_var(self, direct=False):
-        return os.getenv("TEST_SSH_DIRECT", direct)
+        return direct or os.getenv("TEST_SSH_DIRECT") is not None
```

Comment:
That changes the semantics of `TEST_SSH_DIRECT= program` though, i. e. empty string now counts as True. I'm not sure how important that is, but usually an empty value in shell counts as false-y. Would you mind doing `bool(getenv(..))` instead?

Reply:
Alright, simplified it!